rero_ils/modules/notifications/utils.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Notification utils."""
+import ciso8601
+from flask_security.utils import config_value
+from invenio_mail.api import TemplatedMessage
+from invenio_mail.tasks import send_email
+
+from rero_ils.modules.documents.api import Document
+from rero_ils.modules.item_types.api import ItemType
+from rero_ils.modules.items.api import Item
+from rero_ils.modules.locations.api import Location
+from rero_ils.modules.patrons.api import Patron
+
+
+def send_notification_to_location(loan, item, location):
+    """Send a notification to the location defined email.
+
+    :param loan: the loan to be parsed
+    :param location: the location to inform
+    """
+
+    def build_context():
+        document_pid = Item.get_document_pid_by_item_pid(loan.item_pid)
+        document = Document.get_record_by_pid(document_pid)
+        pickup_location = Location.get_record_by_pid(
+            loan.get('pickup_location_pid')
+        )
+        patron = Patron.get_record_by_pid(loan.patron_pid)
+        ctx = {
+            'loan': loan.replace_refs().dumps(),
+            'item': item.replace_refs().dumps(),
+            'document': document.replace_refs().dumps(),
+            'pickup_location': pickup_location,
+            'item_location': location.dumps(),
+            'patron': patron
+        }
+        ctx['pickup_location']['library'] = pickup_location.get_library()
+        ctx['item']['item_type'] = \
+            ItemType.get_record_by_pid(item.item_type_pid)
+        titles = [title for title in ctx['document'].get('title', [])
+                  if title['type'] == 'bf:Title']
+        ctx['document']['title_text'] = \
+            next(iter(titles or []), {}).get('_text')
+        from ..documents.views import create_title_responsibilites
```

Comment:
At the top of the file or at lease the top of the function

Reply:
done